synapse/handlers/pagination.py

```diff
@@ -239,14 +174,6 @@ async def purge_history_for_rooms_in_range(
         for room_id, retention_policy in rooms.items():
             logger.info("[purge] Attempting to purge messages in room %s", room_id)
 
-            if room_id in self._purges_in_progress_by_room:
-                logger.warning(
-                    "[purge] not purging room %s as there's an ongoing purge running"
-                    " for this room",
-                    room_id,
-                )
-                continue
```

Comment:
Do we not need to keep a check like this somewhere?

Reply:
I am not sure. For handling retention I am purposely avoiding using the task scheduler and use `run_as_background_process` instead.
This is because it is a reconciliation loop and purging will happen on next periodic run, so I don't think we need those to be resumable.

I was thinking about just let the global purge room lock handling this. On the other hand, since it is a reconciliation loop, I also don't see real harm if we just skip a room because it is currently purge-busy.
I am going to add the check in case of.